src/python/pants/core/target_types.py

```diff
@@ -52,11 +52,11 @@ class Resources(Target):
 # -----------------------------------------------------------------------------------------------
 
 
-class GenericTarget(Target):
+class TargetAlias(Target):
     """A generic target with no specific target type.
 
     This is useful for aggregate targets: https://www.pantsbuild.org/target_aggregate.html.
     """
 
     alias = "target"
```

Comment:
Thoughts on renaming this? `target` isn't very descriptive. Maybe `target_alias`?

Reply:
We might also want to rename so that 1.0 users are more aware this behavior is different now.

Reply:
I ended up removing the change to expand `TargetAlias`, since it was no longer relevant to the implementation here. Would you prefer that add I back expansion, or revert this rename? I still think that we should consider doing it automatically in expansion.

Reply:
cc @benjyw. What do you think? You mentioned that it would be really neat to have this feature where you can run `./pants test $target_alias` and it runs on all its direct deps. While not in scope for 2.0, it's trivial for us to implement now.

My main concern is what to name it and how to explain the behavior changes for 1.0 users - the feature overall sounds like a good one.